lib/sht4x/measurement.ex

```diff
@@ -9,14 +9,26 @@ defmodule SHT4X.Measurement do
     field(:dew_point_c, number)
     field(:humidity_rh, number, enforce: true)
     field(:temperature_c, number, enforce: true)
+    field(:raw_reading_temperature, integer, enforce: true)
+    field(:raw_reading_humidity, integer, enforce: true)
     field(:timestamp_ms, integer, enforce: true)
   end
 
-  @spec from_raw(<<_::48>>) :: t()
-  def from_raw(<<raw_t::16, _crc1, raw_rh::16, _crc2>>) do
+  @spec from_raw(<<_::48>>, Keyword.t()) :: t()
+  def from_raw(<<raw_t::16, _crc1, raw_rh::16, _crc2>>, opts) do
+    temperature_compensation_func =
+      Keyword.get(opts, :temperature_compensation_func, fn v -> v end)
+
+    humidity_compensation_func = Keyword.get(opts, :humidity_compensation_func, fn v -> v end)
```

Comment:
I think there should be one function that does everything since I don't think there's a case where you'd want to do one and not the other. I'd also change it to take the struct since the timestamp will be needed. I realize that it's easy to get a timestamp, but it seems better to guarantee that everyone looking at the data is using the same timestamp value.

Also, just a super minor thing, `fn v -> v end` can be replaced by `&Function.identity/1`.

Reply:
Thanks for the feedback! I agree with you suggestions, having seen what the compensation function looks like from Sensirion. Also I didn't know about `Function.identity/1`! That's neat.